### PR TITLE
Enrich customer tagging

### PR DESCRIPTION
**Pull request by olsiqose-nosto**

### CODE REVIEW

File: app/code/community/Nosto/Tagging/Block/Customer.php [hunk 1]
```diff
@@ -115,6 +115,15 @@ public function getNostoCustomer()
             $nostoCustomer->setDateOfBirth(DateTime::createFromFormat("Y-m-d H:i:s", $dateOfBirth));
         }
         $nostoCustomer->setMarketingPermission($emailHelper->isOptedIn($email));
+        $customerAddress = $customer->getPrimaryShippingAddress();
+        if ($customerAddress) {
+            $nostoCustomer->setCity($customerAddress->getCity());
+            $nostoCustomer->setStreet($customerAddress->getStreet()[0]);
```
Comment (supercid) [inline app/code/community/Nosto/Tagging/Block/Customer.php]: We have two lines for the street address, might wanna concatenate them?

File: app/code/community/Nosto/Tagging/Block/Customer.php [hunk 2]
```diff
@@ -115,6 +115,15 @@ public function getNostoCustomer()
             $nostoCustomer->setDateOfBirth(DateTime::createFromFormat("Y-m-d H:i:s", $dateOfBirth));
         }
         $nostoCustomer->setMarketingPermission($emailHelper->isOptedIn($email));
+        $customerAddress = $customer->getPrimaryShippingAddress();
+        if ($customerAddress) {
+            $nostoCustomer->setCity($customerAddress->getCity());
```
Comment (supercid) [inline app/code/community/Nosto/Tagging/Block/Customer.php]: This method `getCity()` is not documented, might wanna try-catch and disable the inspection for it

File: Jenkinsfile [hunk 3]
```diff
@@ -57,7 +57,7 @@ pipeline {
     stage('Phan Analysis') {
       steps {
         catchError {
-          sh "./vendor/bin/phan --config-file=phan.php --output-mode=checkstyle --output=chkphan.xml"
```
Comment (supercid) [inline Jenkinsfile]: Revert this

File: app/code/community/Nosto/Tagging/Block/Customer.php [hunk 4]
```diff
@@ -115,6 +115,19 @@ public function getNostoCustomer()
...
         }
         $nostoCustomer->setMarketingPermission($emailHelper->isOptedIn($email));
+        $customerAddress = $customer->getPrimaryShippingAddress();
+        if ($customerAddress) {
+            try {
+                $nostoCustomer->setCity($customerAddress->getCity());
+            } catch (Exception $e) {
+                Mage::logException($e);
+            }
+            $nostoCustomer->setStreet($customerAddress->getStreet()[0].' '.$customerAddress->getStreet()[1]);
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Block/Customer.php]: Can `$customerAddress->getStreet()` return something else that an array? I'd check the type just to be safe

File: app/code/community/Nosto/Tagging/Block/Customer.php [hunk 5]
```diff
@@ -115,6 +115,19 @@ public function getNostoCustomer()
             $nostoCustomer->setDateOfBirth(DateTime::createFromFormat("Y-m-d H:i:s", $dateOfBirth));
         }
         $nostoCustomer->setMarketingPermission($emailHelper->isOptedIn($email));
+        $customerAddress = $customer->getPrimaryShippingAddress();
+        if ($customerAddress) {
+            try {
+                $nostoCustomer->setCity($customerAddress->getCity());
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Block/Customer.php]: Why only this is inside `try`? Should we add address & region setters inside `try` block as well

File: app/code/community/Nosto/Tagging/Block/Customer.php [hunk 6]
```diff
@@ -115,6 +115,19 @@ public function getNostoCustomer()
             $nostoCustomer->setDateOfBirth(DateTime::createFromFormat("Y-m-d H:i:s", $dateOfBirth));
         }
         $nostoCustomer->setMarketingPermission($emailHelper->isOptedIn($email));
+        $customerAddress = $customer->getPrimaryShippingAddress();
+        if ($customerAddress) {
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Block/Customer.php]: Can we check the type here? 
```suggestion
        if ($customerAddress instanceof Mage_Customer_Model_Address) {
```

File: app/code/community/Nosto/Tagging/Block/Customer.php [hunk 7]
```diff
@@ -115,6 +115,19 @@ public function getNostoCustomer()
...
+            try {
+                $nostoCustomer->setCity($customerAddress->getCity());
+            } catch (Exception $e) {
+                Mage::logException($e);
+            }
+            $nostoCustomer->setStreet($customerAddress->getStreet()[0].' '.$customerAddress->getStreet()[1]);
+            $customerRegion = $customerAddress->getRegion();
```
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Block/Customer.php]: Redundant variable assignment

File: app/code/community/Nosto/Tagging/Block/Customer.php (hunk 4 above)
Comment (supercid) [reply app/code/community/Nosto/Tagging/Block/Customer.php]: Just found out that you can pass the line as argument, like:
```php
$customerAddress->getStreet(-1);
```

File: app/code/community/Nosto/Tagging/Block/Customer.php (hunk 7 above)
Comment (olsiqose-nosto) [reply app/code/community/Nosto/Tagging/Block/Customer.php]: Assigned value to variable because it is used more than once

**Pull request merged**